### PR TITLE
[in_app_purchase] Add expiration date to Transaction

### PR DESCRIPTION
**Pull request by feinstein**
Closes https://github.com/flutter/flutter/issues/158226

Exposes the Transaction's expiration date, so this information can be used on the dart side.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/StoreKit2/StoreKit2Translators.swift [hunk 1]
```diff
@@ -188,14 +188,15 @@ extension Product.PurchaseResult {
 extension Transaction {
   func convertToPigeon(restoring: Bool = false) -> SK2TransactionMessage {
 
-    let dateFromatter: DateFormatter = DateFormatter()
-    dateFromatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
+    let dateFormatter: DateFormatter = DateFormatter()
```
Comment (hellohuanlin) [inline packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/StoreKit2/StoreKit2Translators.swift:191]: nit: `let dateFormatter = DateFormatter()`

**Review by hellohuanlin — APPROVED** [review]
Can you provide more details in PR description?

**Review by LouiseHsu — APPROVED** [review]
lgtm! just a couple nits about the pubspec.

**Comment by feinstein** [comment]
> Can you provide more details in PR description?

Done.

File: packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/StoreKit2/StoreKit2Translators.swift (hunk 1 above)
Comment (feinstein) [reply packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/StoreKit2/StoreKit2Translators.swift:191]: Sure, will update shortly. Just keep in mind I am not a swift dev, so snippets like this one are really helpful!

File: packages/in_app_purchase/in_app_purchase/example/pubspec.yaml [hunk 2]
```diff
@@ -28,3 +28,8 @@ dev_dependencies:
 
 flutter:
   uses-material-design: true
+
```
[the remark this answers is not in the document]
Comment (feinstein) [reply packages/in_app_purchase/in_app_purchase/example/pubspec.yaml:31]: I was under the impression, from the CONTRIBUITING docs, that we first send a PR with dependencies overrides, created by the repo tool, so all packages and examples can be tested with the latest changes. Then after the PR is approved I remove those in a new PR: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changing-federated-plugins But maybe I got it wrong?

File: packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/StoreKit2/StoreKit2Translators.swift (hunk 1 above)
Comment (feinstein) [reply packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/StoreKit2/StoreKit2Translators.swift]: Done

File: packages/in_app_purchase/in_app_purchase/example/pubspec.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (stuartmorgan) [reply packages/in_app_purchase/in_app_purchase/example/pubspec.yaml:31]: > I was under the impression, from the CONTRIBUITING docs, that we first send a PR with dependencies overrides, created by the repo tool

That's for changes that span multiple packages; this PR only affects one package.

[the remark this answers is not in the document]
Comment (feinstein) [reply packages/in_app_purchase/in_app_purchase/example/pubspec.yaml:31]: Ah, I see, thanks, I thought it was needed to test the packages, using local references.

File: packages/in_app_purchase/in_app_purchase/pubspec.yaml [hunk 3]
```diff
@@ -42,3 +42,8 @@ screenshots:
     path: doc/iap_ios.gif
   - description: 'Example of in-app purchase on android'
     path: doc/iap_android.gif
+
+# FOR TESTING AND INITIAL REVIEW ONLY. DO NOT MERGE.
+# See https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changing-federated-plugins
```
[the remark this answers is not in the document]
Comment (feinstein) [reply packages/in_app_purchase/in_app_purchase/pubspec.yaml]: Done

File: packages/in_app_purchase/in_app_purchase/example/pubspec.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (feinstein) [reply packages/in_app_purchase/in_app_purchase/example/pubspec.yaml]: Done

**Review by jmagman — COMMENTED** [review]
Dang merge conflict with https://github.com/flutter/packages/pull/7964. Can you regenerate the files as needed and bump the version and update CHANGELOG again? Sorry I know that's annoying.

**Comment by feinstein** [comment]
> Dang merge conflict with #7964. Can you regenerate the files as needed and bump the version and update CHANGELOG again? Sorry I know that's annoying.

It's ok, part of the life of a dev

**Comment by stuartmorgan** [comment]
Reviewers, is this ready to `autosubmit`, or are there pending final reviews?

**Comment by LouiseHsu** [comment]
> Reviewers, is this ready to `autosubmit`, or are there pending final reviews?

LGTM! @feinstein you can add the `autosubmit` label when you're ready to merge.

**Comment by stuartmorgan** [comment]
@LouiseHsu Only project members can add labels; the final reviewer should add the label for non-member contributions.

It looks like the PR that I just landed re-broke the version change; I'll fix that and add the label.

**Comment by stuartmorgan** [comment]
(Reviewers, please remember as part of review to check that the type of the version increment follows semver; new API additions are minor version changes, not bugfixes.)

**Comment by feinstein** [comment]
@stuartmorgan I though that would be the case, but when I saw past entries in the changelog the last ones were just incrementing patch, so I followed this. Thanks for the review and fixing it :).

**Pull request merged**